scripts/binary/binary-integrity-check-source.js

```diff
@@ -0,0 +1,168 @@
+const OrigError = Error
+const captureStackTrace = Error.captureStackTrace
+const toString = Function.prototype.toString
+const callFn = Function.call
+
+// eslint-disable-next-line no-unused-vars
+const stackIntegrityCheck = function stackIntegrityCheck (options) {
+  const originalStackTraceLimit = OrigError.stackTraceLimit
+  const originalStackTrace = OrigError.prepareStackTrace
+
+  OrigError.stackTraceLimit = Infinity
+
+  OrigError.prepareStackTrace = function (_, stack) {
+    return stack
+  }
+
+  const tempError = new OrigError
+
+  captureStackTrace(tempError, arguments.callee)
+  const stack = tempError.stack.filter((frame) => !frame.getFileName().startsWith('node:internal') && !frame.getFileName().startsWith('node:electron'))
+
+  OrigError.prepareStackTrace = originalStackTrace
+  OrigError.stackTraceLimit = originalStackTraceLimit
+
+  if (stack.length !== options.stackToMatch.length) {
+    throw new Error(`Integrity check failed with expected stack length ${options.stackToMatch.length} but got ${stack.length}`)
+  }
+
+  for (let index = 0; index < options.stackToMatch.length; index++) {
+    const expectedFunctionName = options.stackToMatch[index].functionName
+    const actualFunctionName = stack[index].getFunctionName()
+    const expectedFileName = options.stackToMatch[index].fileName
+    const actualFileName = stack[index].getFileName()
+
+    if (expectedFunctionName && actualFunctionName !== expectedFunctionName) {
```

Comment:
should we throw if expectedFunctionName or expectedFunctionName are undefined? can we get in this situation?

Reply:
looking below looks like expectedFunctionName depends on the file

Reply:
We control what's passed in here so we shouldn't be in that situation.